lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/backend_impl.cpp

```diff
@@ -69,7 +69,16 @@ class TSBackendImpl : public torch::lazy::BackendImplInterface {
       const at::Tensor& tensor, const torch::lazy::Shape& shape,
       const torch::lazy::BackendDevice& device) const override {
     at::TensorOptions options = tensor.options().device(default_device_type_.c10Type());
-    return std::make_shared<TSData>(tensor.to(options), shape, device);
+    if (tensor.device().type() == default_device_type_.c10Type()) {
```

Comment:
wouldn't this be unsafe on CPU to CPU, or will it be sync on CPU regardless of `non_blocking` ?

Reply:
yea that's a good question. i assumed cpu is always blocking but i need to confirm that.

i guess i could just make it a special case for cuda and be sure.

Reply:
yes please!!! Could we please stick in  `&& tensor.device().type() == c10::kCUDA` in the first conditional for now until we confirm it works for CPU as well.